numpyro/infer/mcmc.py

```diff
@@ -247,11 +253,11 @@ def init_kernel(init_params,
                                             target_accept_prob=target_accept_prob,
                                             find_reasonable_step_size=find_reasonable_ss)
 
-        rng_key_hmc, rng_key_wa = random.split(rng_key)
+        rng_key_hmc, rng_key_wa, rng_key_momentum = random.split(rng_key, 3)
```

Comment:
This split gives some failing tests. So I changed their thresholds a bit.

Reply:
can you please explain further?

Reply:
This issue happens at `hmc` function (not relevant to this PR). We missed a split to generate a key for the initial `r`. Because this line will affect the following random generator numbers, so some strict tests will fail.